paper-server/patches/sources/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -1540,6 +1540,28 @@
                  this.doPush(entity1);
              }
          }
+@@ -2954,6 +_,21 @@
+     protected void checkAutoSpinAttack(AABB boundingBoxBeforeSpin, AABB boundingBoxAfterSpin) {
+         AABB aabb = boundingBoxBeforeSpin.minmax(boundingBoxAfterSpin);
+         List<Entity> entities = this.level().getEntities(this, aabb);
++        // Paper start - allow customising which entities are attacked
++        entities.removeIf(entity -> {
++            // Filter out non-living entities properly without breaking
++            // horizontal collisions
++            if (!(entity instanceof LivingEntity)) return true;
```

Comment:
I am a bit confused by this line. It seems we are eagerly dropping living entities, which could yield in entities becoming empty when previously it would not have (e.g. entities only containing non living entities).
This would then prompt a check for horizontal collision where it would not in vanilla.

Tho tbf, I am also semi confused by mojangs logic anyway.

I think the most "vanilla friendly" thing here is to check for horizonal collision only if the list itself is empty or if all entities inside are either invisible to the player or had their event cancelled.
That will require a bit of rewiring of the else-if, presumably with a quick boolean flag, but yea.

Reply:
The annoying part is that Mojang's logic is just wrong how I see it, I assume they intend to check for a horizontal collision if no spin attack occurred, so that's the logic that I match. This just replaces the living entity check inside the loop basically, but doesn't remove that to not edit extra lines.